cmd/serverless/debug.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const DD_PREFIX = "DD_"
+
+func outputDatadogEnvVariablesForDebugging() {
+	log.Debug(buildDebugString())
+}
+
+func buildDebugString() string {
+	var sb strings.Builder
+	envMap := getEnvVariableToObfuscate()
+	sb.WriteString("Datadog environment variables: ")
+	allEnv := os.Environ()
+	sort.Strings(allEnv)
+	for _, pair := range allEnv {
+		if strings.HasPrefix(pair, DD_PREFIX) {
+			sb.WriteString(obfuscatePairIfNeeded(pair, envMap))
+			sb.WriteString("|")
+		}
+	}
+	return sb.String()
+}
+
+func obfuscatePairIfNeeded(pair string, envMap map[string]bool) string {
```

Comment:
I see what you were going for with dependency injection here, but it might be simpler to just use the hardcoded env variables to obfuscate here.